### PR TITLE
Add unit testcases to eos_vlans

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY

Added Unit testcases for the resource module eos_vlans

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
/home/gosriniv/ansible/test/units/modules/network/eos/test_eos_vlans.py

### CODE REVIEW

File: test/units/modules/network/eos/test_eos_vlans.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        self.mock_get_resource_connection_facts.stop()
+        self.mock_edit_config.stop()
+        self.mock_get_config.stop()
+        self.mock_load_config.stop()
+        self.mock_execute_show_command.stop()
+
+    def load_fixtures(self, commands=None, transport='cli'):
+        file_cmd = load_fixture('eos_vlan_config.cfg').split()
+        #       comprehension failing in zuul with syntax error
+        #       file_cmd_dict = {file_cmd[i]: file_cmd[i + 1] for i in range(0, len(file_cmd), 2)}
```
Comment (Qalthos) [inline test/units/modules/network/eos/test_eos_vlans.py]: Yeah, we still can't use dict comprehensions until 2.6 is formally dropped, sorry.

File: test/units/modules/network/eos/test_eos_vlans.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+    def test_eos_vlan_default_idempotent(self):
+        self.execute_show_command.return_value = load_fixture('eos_vlan_config.cfg')
+        set_module_args(dict(
+            config=[dict(
+                vlan_id=10,
+                name="ten"
+            )]
+        ))
+        commands = ['vlan 10', 'name ten']
```
Comment (Qalthos) [inline test/units/modules/network/eos/test_eos_vlans.py]:
```suggestion
```

File: test/units/modules/network/eos/test_eos_vlans.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+    def test_eos_vlan_merged_idempotent(self):
+        self.execute_show_command.return_value = load_fixture('eos_vlan_config.cfg')
+        set_module_args(dict(
+            config=[dict(
+                vlan_id=10,
+                name="ten"
+            )], state="merged"
+        ))
+        commands = ['vlan 10', 'name ten']
```
Comment (Qalthos) [inline test/units/modules/network/eos/test_eos_vlans.py]:
```suggestion
```

File: test/units/modules/network/eos/test_eos_vlans.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    #        self.assertIn("we were unable to convert to int", result['msg'])
+
+    def test_eos_vlan_state_datatype(self):
+        set_module_args(dict(
+            config=[dict(
+                vlan_id=30,
+                state=10
+            )]
+        ))
+        commands = ['vlan 30', 'state 10']
```
Comment (Qalthos) [inline test/units/modules/network/eos/test_eos_vlans.py]:
```suggestion
```

File: test/units/modules/network/eos/test_eos_vlans.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+    # Bug exists. Will be uncommented once the bug is fixed
+
+    #    def test_eos_vlan_id_datarange(self):
+    #        set_module_args(dict(
+    #            config=[dict(
+    #                vlan_id=20000
+    #            )]
+    #        ))
+    #        commands = ['vlan outofrange']
```
Comment (Qalthos) [inline test/units/modules/network/eos/test_eos_vlans.py]:
```suggestion
```

File: test/units/modules/network/eos/test_eos_vlans.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        commands = ['vlan 10', 'no name']
+        self.execute_module(changed=True, commands=commands)
+
+    def test_eos_vlan_id_datatype(self):
+        set_module_args(dict(
+            config=[dict(
+                vlan_id="thirty"
+            )]
+        ))
+        commands = ['vlan thirty']
```
Comment (Qalthos) [inline test/units/modules/network/eos/test_eos_vlans.py]:
```suggestion
```

File: test/units/modules/network/eos/test_eos_vlans.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+    def test_eos_vlan_overridden_idempotent(self):
+        self.execute_show_command.return_value = load_fixture('eos_vlan_config.cfg')
+        set_module_args(dict(
+            config=[dict(
+                vlan_id=10,
+                name="ten"
+            )], state="overridden"
+        ))
+        commands = ['vlan 10', 'name ten']
```
Comment (Qalthos) [inline test/units/modules/network/eos/test_eos_vlans.py]:
```suggestion
```

File: test/units/modules/network/eos/test_eos_vlans.py [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+    def test_eos_vlan_replaced_idempotent(self):
+        self.execute_show_command.return_value = load_fixture('eos_vlan_config.cfg')
+        set_module_args(dict(
+            config=[dict(
+                vlan_id=10,
+                name="ten"
+            )], state="replaced"
+        ))
+        commands = ['vlan 10', 'name ten']
```
Comment (Qalthos) [inline test/units/modules/network/eos/test_eos_vlans.py]:
```suggestion
```

**Review by Qalthos — CHANGES_REQUESTED** [review]
You don't need to specify commands if you're not going to assert them with `execute_module`.

Is the vlan issue fixed? If not, what is the issue #? I prefer not to commit commented tests without good reason

File: test/units/modules/network/eos/test_eos_vlans.py (hunk 1 above)
Comment (GomathiselviS) [reply test/units/modules/network/eos/test_eos_vlans.py]: Addressed the review comments

Comment (Qalthos) [reply test/units/modules/network/eos/test_eos_vlans.py]: Can you remove the commented comprehensions? It is unfortunate that dict comprehensions aren't available, but having commented code floating around is rarely helpful.

Comment (GomathiselviS) [reply test/units/modules/network/eos/test_eos_vlans.py]: Done!

**Review by Qalthos — APPROVED** [review]

**Pull request merged**